packages/rpc-core/src/lamports.ts

```diff
@@ -0,0 +1,19 @@
+// FIXME(solana-labs/solana/issues/30341) Beware that any value above 9007199254740991 may be
+// truncated or rounded because of a downcast to JavaScript `number` between your calling code and
+// the JSON-RPC transport.
+export type LamportsUnsafeBeyond2Pow53Minus1 = bigint & { readonly __brand: unique symbol };
+
+// Largest possible value to be represented by a u64
+const maxU64Value = 2n ** 64n - 1n;
```

Comment:
Prefer to bake this rather than to do math in the module constructor.
```suggestion
const maxU64Value = 18446744073709551615n; // 2n ** 64n - 1n
```